kong/plugins/ai-proxy/handler.lua

```diff
@@ -0,0 +1,133 @@
+local _M = {}
+
+-- imports
+local ai_shared = require("kong.plugins.ai-proxy.drivers.shared")
+local fmt = string.format
+local kong_utils = require("kong.tools.utils")
+local kong_meta = require "kong.meta"
+--
+
+_M.PRIORITY = 770
+_M.VERSION = kong_meta.version
+
+local INTERNAL_SERVER_ERROR_TEMPLATE = '{"error":{"message":"%s"}}'
+
+local function bad_request(msg)
+  kong.log.warn(msg)
+  return kong.response.exit(400, { message = msg, error = true })
+end
+
+local function internal_server_error(msg)
+  kong.log.err(msg)
+  return kong.response.exit(500, { message = msg, error = true })
+end
+
+function _M:header_filter(conf)
+  if not kong.ctx.shared.skip_response_transformer then
+    -- clear shared restricted headers
+    for i, v in ipairs(ai_shared.clear_response_headers.shared) do
+      kong.response.clear_header(v)
+    end
+
+    -- only act on 200 in first release - pass the unmodifed response all the way through if any failure
+    if kong.response.get_status() == 200 then
+      local ai_driver = require("kong.plugins.ai-proxy.drivers." .. conf.model.provider)
+      local route_type = conf.route_type
+      
+      local response_body = kong.service.response.get_raw_body()
+
+      if response_body then
+        local is_gzip = kong.response.get_header("Content-Encoding") == "gzip"
+        if is_gzip then
+          response_body = kong_utils.inflate_gzip(response_body)
+        end
+
+        local new_response_string, err = ai_driver.from_format(response_body, conf.model, route_type)
+        if err then
+          ngx.status = 500
+          kong.ctx.plugin.parsed_response = fmt(INTERNAL_SERVER_ERROR_TEMPLATE, err)
+        end
+        if new_response_string then
```

Comment:
let's combine these into a single if/elseif

```suggestion
        
        elseif new_response_string then
```